docs/proposals/removal-tekton-resources.md

```diff
@@ -0,0 +1,269 @@
+- For `spec.sources`, arbitrary resources can emit `source-${spec.sources[i].name}-${resultName}`. For example, assuming the user's Build contains a HTTP source named `license-text` and the HTTP source would be emitting a result named `size`, then the name of Tekton result will be `source-license-text-size`. Note: this proposal only means to define the naming pattern for the results from `spec.sources`. The decision on which results we introduce for the HTTP sources and for future source types should be covered elsewhere.
+
+The above logic accepts the risk of conflicts if a user's Build contains an item in `spec.sources` that uses the name `default`. The risk is accepted based on the assumption that `spec.source` will eventually be removed and that we currently do not add any result for the HTTP sources.
+
+Tekton results work by expecting the container to write the value of the result to a file. Tekton then puts the content of those files into the termination message of the container and from there, it puts them into the TaskRun's status. Termination messages have a maximum length of 4 kB. This also limits how many results one can fill and how long those values can be. This must be considered in the future when we extend our results.
+
+The results for the sources will be filled by our own code as described later. A build strategy author must not do anything to fill them.
+
+For the output image, we will also provide two more system-defined results: `image-digest` and `image-size`. As the image building process is part of the build strategy, only the build strategy author can have the knowledge about where to get those values from and subsequently to write them to files. The build strategy author will access the path for the result files using Tekton's replacement variables for results, in this case: `$(results.image-digest.path)` and `$(results.image-size.path)`. This will be documented in our build strategy documentation. A build strategy author is not required to provide a value for those results. In case values are not provided, they will not be present in the TaskRun status.
+
+In our sample build strategies, we should support these two results as much as possible:
+
+- `ko` supports to write an OCI image manifest file using the `--oci-layout-path` argument. The strategy needs to write that manifest to a temporary location, use a tool to extract the digest and size from the index.json file, and write this data to the result files.
+- Kaniko supports to write an OCI image manifest file using the `--oci-layout-path` argument. Kaniko is based on scratch, as such, we cannot extend the existing build-and-push step to use some tool and run some extraction logic. We need to add an extra step. For this, we need to add volumeMounts to write the temporary file in the `build-and-push` step, and consume it in the following one to extract and store the results.
```

Comment:
We could use the Kaniko debug image, which includes a shell: https://github.com/GoogleContainerTools/kaniko/blob/master/README.md#debug-image

Reply:
Hm, that might not work. Kaniko works by running all steps of the Dockerfile in its own container. When loading the `FROM` image, this overwrites the root file system of the Kaniko image which normally is not there as it is based on scratch. So, after Kaniko finished, you have no guarantee on what is still there (I might have missed it, but I think `/busybox` is not excluded when Kaniko for example cleans up the whole file system, beside /kaniko, between two stages). I think the main purpose of the debug image is to start the container with a shell to be able to debug things and manually start Kaniko inside the container. I think we would skate on thin ice to use debug to be able to run something after Kaniko finished.